room.h

```diff
@@ -90,6 +90,7 @@ namespace QMatrixClient
             QList<User*> membersLeft() const;
 
             Q_INVOKABLE QList<User*> users() const;
+			Q_INVOKABLE QStringList usernames() const;
```

Comment:
It should; try to add `Q_DECLARE_METATYPE(User*)` to `main.cpp` (there are already a couple of custom pointer types registered in that way).